lib/src/main/java/grpcbridge/route/BodyParser.java

```diff
@@ -69,7 +78,29 @@ public RpcMessage extract(HttpRequest request) {
         return request.getBody()
                 .map(requestBody -> {
                     if (bodyExtractor == null) {
-                        return ProtoJson.parse(request, blank.toBuilder());
+                        String contentType = request.getHeaders().get(
+                                Metadata.Key.of("content-type",
+                                Metadata.ASCII_STRING_MARSHALLER)
+                        );
+                        final Parser parser;
+                        if (contentType == null) {
```

Comment:
Instead of doing this logic everywhere I'd have a wrapper around List<Parser> class. The wrapper then would be aware of the default parser and return it when nothing matches.

Reply:
Agreed, that will be much cleaner, I'll make a utility that externalizes this logic